### DISCUSSION TITLE
Is there a way to generate prism syntax file?

### DISCUSSION THREAD

**Discussion by ymc9** [eclipse-langium/langium#785] [Q&A]
Hi guys,

I'm trying to get my language syntax highlighting on a [docsify](https://docsify.js.org/) site, which uses Prism for syntax highlighting. Just wondering if there's a way of generating a Prism grammar definition from Langium file.

Thanks!

**Comment by msujew** [comment]
Hey @ymc9,

it might require a bit of work, but you could fork the [langium-cli](https://github.com/langium/langium/tree/main/packages/langium-cli) and add a new highlighter similar to the ones we already have, see [here](https://github.com/langium/langium/tree/main/packages/langium-cli/src/generator/highlighting). If you like, you could also contribute it back to our repo. I'm sure a few people could profit from this feature.

**Reply by ymc9** [reply]
Got it, and thanks!

I'll be happy to contribute. Need to prep myself about Prism first 😄.

**Reply by msujew** [reply]
@ymc9 I came around to build Prism integration for the generator. See https://github.com/langium/langium/pull/995. See it in action over at this PR https://github.com/langium/langium-website/pull/140.

**Reply by ymc9** [reply]
Awesome! I'm using Docusaurus for my doc site, and it seems that I can setup a custom prism syntax. Let me try it out. Many thanks for working on this!

**Reply by ymc9** [reply]
Hi @msujew , it seems the PR hasn't been merged yet. Will it be included in the next release 😄?

**Reply by montymxb** [reply]
@ymc9 I'll be taking a look at the generator PR sometime today, we can probably expect to have it added in soon afterwards 👍 .
